### PR TITLE
Posix wait

### PR DESCRIPTION
**Pull request by ikbenlike**
This PR extends the functionality of `wait` and `waitpid` to also except NULL as argument, since the standard specifies this is possible. It also adds the glibc `error` header, and implements the functions therein.

### CODE REVIEW

File: options/glibc/generic/error.cpp [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+unsigned int error_message_count = 0;
+int error_one_per_line = 0;
+void (*error_print_progname)(void) = NULL;
+
+void error(int status, int errnum, const char *format, ...) {
+	va_list args;
+	va_list copied;
+	va_start(args, format);
+	va_copy(copied, args);
+	va_end(args);
```
Comment (ArsenArsen) [inline options/glibc/generic/error.cpp:17]: I use va_* APIs maybe once a year at this point, so not entirely sure if I'm right, but: isn't the copy here redundant?

Comment (qookei) [reply options/glibc/generic/error.cpp:17]: As far as I know a copy is only needed if you want to iterate through the same varargs twice (e.g. when calling 2 functions that take a `va_list` with the same list).

Comment (ikbenlike) [reply options/glibc/generic/error.cpp:17]: I think I remember something about needing to copy the list of reusing it, but I'm not 100% sure and this API isn't really my strong suit. If this isn't required, would I simply be able to initialize the list and pass it on to `vfprintf`?

Comment (qookei) [reply options/glibc/generic/error.cpp:17]: Yeah if you want to use the same list twice you need to make a copy, but in this case you only use it to pass it on to `vfprintf`, in which case you can just pass the original one through.

Comment (ikbenlike) [reply options/glibc/generic/error.cpp:17]: Makes sense; I'll change this later.

Comment (ikbenlike) [reply options/glibc/generic/error.cpp]: This has been corrected.

File: tests/glibc/error.py [hunk 2]
```diff
@@ -0,0 +1,7 @@
+import subprocess
+import sys
+from pyexpect import expect
+
+completed = subprocess.run([sys.argv[1]], check=True, capture_output=True)
```
Comment (ArsenArsen) [inline tests/glibc/error.py:5]: this could be just `check_output` instead of run

File: options/glibc/include/error.h [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+
+extern unsigned int error_message_count;
+extern int error_one_per_line;
+extern void (*error_print_progname)(void);
+	
+#ifdef __cplusplus
+}
+#endif
+	
+#endif // _ERROR_H
```
Comment (ArsenArsen) [inline options/glibc/include/error.h:21]: we can't use these comments in public headers

**Review by ArsenArsen — CHANGES_REQUESTED** [review]

File: options/glibc/include/error.h (hunk 3 above)
Comment (ikbenlike) [reply options/glibc/include/error.h:21]: why not?

Comment (qookei) [reply options/glibc/include/error.h:21]: C89 doesn't support single-line comments

File: tests/glibc/error.py (hunk 2 above)
Comment (ikbenlike) [reply tests/glibc/error.py]: This change has been made

File: options/glibc/include/error.h (hunk 3 above)
Comment (ikbenlike) [reply options/glibc/include/error.h]: I removed it

**Review by ArsenArsen — APPROVED** [review]

File: options/glibc/include/error.h (hunk 3 above)
Comment (Dennisbonke) [reply options/glibc/include/error.h]: `/* _ERROR_H */` is preferred.

Comment (ikbenlike) [reply options/glibc/include/error.h]: This change has been made.

File: options/glibc/generic/error.cpp [hunk 4]
```diff
@@ -0,0 +1,67 @@
...
+void error(int status, int errnum, const char *format, ...) {
+	va_list args;
+	va_start(args, format);
+
+	error_message_count++;
+	
+	fflush(stdout);
+	if(error_print_progname) {
+		error_print_progname();
+	}
```
Comment (Dennisbonke) [inline options/glibc/generic/error.cpp:21]: Please put the `else` on the same line as the closing bracket.

File: options/glibc/generic/error.cpp [hunk 5]
```diff
@@ -0,0 +1,67 @@
...
+	if(!(last_line == linenum && error_one_per_line && !first_call)) {
+		first_call = false;
+		last_line = linenum;
+		error_message_count++;
+
+		fflush(stdout);
+		if(error_print_progname) {
+			error_print_progname();
+		}
+		else {
```
Comment (Dennisbonke) [inline options/glibc/generic/error.cpp:52]: Please put the `else` on the same line as the closing bracket.

File: options/posix/generic/sys-wait-stubs.cpp [hunk 6]
```diff
@@ -14,11 +14,15 @@ int waitid(idtype_t, id_t, siginfo_t *, int) {
...
 pid_t waitpid(pid_t pid, int *status, int flags) {
 	pid_t ret;
+	int tmp_status = 0;
 	MLIBC_CHECK_OR_ENOSYS(mlibc::sys_waitpid, -1);
-	if(int e = mlibc::sys_waitpid(pid, status, flags, NULL, &ret); e) {
+	if(int e = mlibc::sys_waitpid(pid, &tmp_status, flags, NULL, &ret); e) {
 		errno = e;
 		return -1;
 	}
+	if(status){
```
Comment (Dennisbonke) [inline options/posix/generic/sys-wait-stubs.cpp:23]: Missing space between `)` and `{`.

**Review by Dennisbonke — CHANGES_REQUESTED** [review]
Code looks good, some style issues were found.

**Review by Dennisbonke — APPROVED** [review]
LGTM

**Pull request merged**